### PR TITLE
Support patchset filtering

### PR DESCRIPTION
**Pull request by spbnick**
This is a work-in-progress support for filtering patchsets. Nothing here was tested, filtering implementation itself is missing, as well as adding cover letter URL to patchset summaries.

### CODE REVIEW

**Comment by spbnick** [comment]
Oh, and database documentation needs to be finished eventually.

File: sktm/db.py [hunk 1]
```diff
@@ -40,15 +40,20 @@ def createdb(self, db):
...
                 );
 
+                /* Patch sources (Patchwork projects) */
                 CREATE TABLE patchsource(
                   id INTEGER PRIMARY KEY,
+                  /* Patchwork base URL */
                   baseurl TEXT,
+                  /* Numeric Patchwork project ID */
                   project_id INTEGER,
+                  /* TODO Document or remove */
```
Comment (veruu) [inline sktm/db.py]: Checking git history, this seems to be a residue of adding events API support and neither `date` nor the `set_event_date` and `get_last_event_date` methods which use it aren't used anymore. Now the `nsince` attribute is used to save these information. I'm fine with removing the column and methods that use it.

**Comment by veruu** [comment]
I finally went through everything and what you have so far looks good! :)

File: sktm/patchwork.py [hunk 2]
```diff
@@ -483,7 +483,7 @@ def get_patch_by_id(self, pid):
 
         return r.json()
 
-    def get_patchsets_by_patch(self, url, db=None, seen=set()):
+    def get_patchsets_by_patch(self, url, seen=set()):
```
Comment (veruu) [inline sktm/patchwork.py]: Do you want to have this in this pull or should I put it into my patchwork pull?

Comment (spbnick) [reply sktm/patchwork.py]: This was just a side-effect of me considering if we should access database in Patchwork, and seeing some stuff there I thought we'd better distance ourselves from that idea. It's not at all needed for adding filtering, so if it's better for you, take it over, and I'll remove it from here :)

File: sktm/db.py (hunk 1 above)
Comment (spbnick) [reply sktm/db.py]: OK, cool :) I'll change this to just `TODO: Remove`.

File: sktm/patchwork.py (hunk 2 above)
Comment (veruu) [reply sktm/patchwork.py]: That's what I thought :) I'll add it to the pull.

**Comment by major** [comment]
No objections here. There are a few cleanups/simplifications we could do later, but the logic looks sound. I haven't tested it myself, though. ;)

**Pull request merged**